docs/guide/tutorial-yii-as-micro-framework.md

```diff
@@ -0,0 +1,119 @@
+# Using Yii as a Micro-framework
+
+Yii can be easily used without features included in basic and advanced templates. In other words, Yii is already a micro-framework. It is not required to have the directory structure provided by templates to work with Yii.
+
+This is especially handy when you do not need all the pre-defined template code like assets or views. One of such cases is building a JSON API. In the following sections will show how to do that.
+
+## Create Project
+
+Create a directory for your project files and change working directory to that path. Commands used in examples are Unix-based but similar commands exist in Windows.
+
+```bash
+mkdir micro-app
+cd micro-app
+```
+
+> Note: Composer knowledge is required to continue. If you don't know how to use composer yet, please take time to read [Composer Guide](https://getcomposer.org/doc/00-intro.md).
+
+Create file `composer.json` under the `micro-app` directory using your favorite editor and add the following:
+
+```json
+{
+    "require": {
+        "yiisoft/yii2": "~2.0.0"
+    },
+    "repositories": [
+        {
+            "type": "composer",
+            "url": "https://asset-packagist.org"
+        }
+    ]
+}
+```
+
+Save the file and run the `composer install` command. This will install the framework with all its dependencies. Alternatively you can skip the manual creation of the `composer.json` file and just run the command `composer require yiisoft/yii2`. This will create the `composer.json` file as well as install the dependencies.
+
+## Create Project Structure
+
+After you have installed the framework, it's time to create an entry point for the application. Entry point is the very first file that will be executed when you try to open your application. For the security reasons, it is recommended to put the entrypoint file in a separate directory and make it a web root. 
+
+Create a `web` directory and put `index.php` inside with the following content:
+
+```php 
+<?php
+
+// comment out the following two lines when deployed to production
+defined('YII_DEBUG') or define('YII_DEBUG', true);
+defined('YII_ENV') or define('YII_ENV', 'dev');
+
+require(__DIR__ . '../vendor/autoload.php');
+require(__DIR__ . '../vendor/yiisoft/yii2/Yii.php');
+
+// you may put this into a separate file
+$config = [
+    'id' => 'micro-app',
+    'basePath' => __DIR__,
```

Comment:
better point it to `dirname(__DIR__)` so it points to the same place as in other applications, this way example code from other places would work out of the box.